vmdb/app/presenters/menu/polymorph_item.rb

```diff
@@ -0,0 +1,7 @@
+module Menu
+  class PolymorphItem < Item
+    def name
+      super.call
```

Comment:
Normal method in the Menu::Item would do it. Not a normal method in the DefaultMenu class. I would have to take the logic that deal with the Provider or Host inside the Menu::Item class or subclass which I don't like.

Or I could pass in a method instead of a proc if proc is what worries you.